drivers/flash/flash_ipuc/flash_ipuc.c

```diff
@@ -645,3 +700,110 @@ bool flash_ipuc_setup_pending(const struct device *dev)
 
 	return false;
 }
+
+struct device *flash_image_ipuc_create(size_t id, struct zcbor_string *encryption_info,
+				       struct zcbor_string *compression_info,
+				       uintptr_t *ipuc_address, size_t *ipuc_size)
+{
+	suit_manifest_role_t role = SUIT_MANIFEST_UNKNOWN;
+	suit_plat_err_t plat_err = SUIT_PLAT_SUCCESS;
+	struct ipuc_context *ctx = NULL;
+	struct device *dev = NULL;
+	size_t count = 0;
+	uint8_t cpu_id = 0;
+	size_t idx = 0;
+#ifdef CONFIG_FLASH_SIMULATOR
+	size_t f_size = 0;
+	uintptr_t base_address = (uintptr_t)flash_simulator_get_memory(NULL, &f_size);
+#else
+	uintptr_t base_address = 0;
+#endif
+
+	for (idx = 0; idx < ARRAY_SIZE(ipuc_imgs); idx++) {
+		if (ipuc_imgs[idx].id == id) {
+			LOG_INF("Found IPUC image %d: (0x%lx, 0x%x)", ipuc_imgs[idx].id,
+				ipuc_imgs[idx].address, ipuc_imgs[idx].size);
+			break;
+		}
+	}
+
+	if (idx >= ARRAY_SIZE(ipuc_imgs)) {
+		LOG_INF("IPUC image with ID %d not found", id);
+		return NULL;
+	}
+
+	if (ipuc_imgs[idx].fdev != NULL) {
+		LOG_WRN("IPUC for image %d already created", id);
+		return NULL;
+	}
+
+	plat_err = suit_ipuc_get_count(&count);
+	if (plat_err != SUIT_PLAT_SUCCESS) {
+		LOG_WRN("Unable to read IPUC count");
+		return NULL;
+	}
+
+	dev = get_free_dev();
+	if ((dev == NULL) || (ipuc_address == NULL) || (ipuc_size == NULL)) {
+		LOG_WRN("Invalid image input arguments: (0x%lx, 0x%lx, 0x%lx)", (uintptr_t)dev,
```

Comment:
This does not seem to be a correct message  for `get_free_dev` failure - the failure results from a too large number of allocated devices, not an incorrect function argument